src/common/WeightedPriorityQueue.h

```diff
@@ -84,7 +84,8 @@ class WeightedPriorityQueue :  public OpQueue <T, K>
       }
       //Get the cost of the next item to dequeue
       unsigned get_cost() const {
-	return lp.begin()->cost;
+        assert(!empty());
```

Comment:
not related to this commit, I'd like to avoid "assert" in fast path in the future changes if not very necessary. Of course, any reasonable assert must be done. To be clarify again, I encourage use assert in necessary path.
@athanatos what do you think?